src/main/java/com/mael/ttt/players/HumanPlayer.java

```diff
@@ -23,24 +23,24 @@ public Mark getMark() {
         return mark;
     }
 
-    private String returnValidCellIndex() {
-        String cellIndex      = gameUI.getInput();
-        String validCellIndex = "[1-9]";
-
-        while (!cellIndex.matches(validCellIndex)) {
+    private String returnValidCellIndex(Board board) {
```

Comment:
Is it possible to split this method to not have it violate the command/query separation?

Additionally this method as well as `returnEmptyCellIndex()` ask the `gameUI` for input.
How could that be streamlined into only one method calling `gameUI.getInput()`.

As it is right now, the responsibility of asking for input feels a bit scattered throughout the class.